### PR TITLE
Fix compatibility with redis-rb 4.2

### PR DESCRIPTION
**Pull request by casperisfine**
Fix https://github.com/redis-store/redis-actionpack/issues/31

The code is a bit convoluted because of the ruby 2.1 requirement. on 2.5+ I could have used Hash#slice.

cc @tubbo @shepmaster

### CODE REVIEW

File: lib/redis/store/namespace.rb [hunk 1]
```diff
@@ -23,8 +23,12 @@ def get(key, options = nil)
...
       end
 
-      def exists(key)
-        namespace(key) { |k| super(k) }
+      def exists(*keys)
+        super(*keys.map { |key| interpolate(key) })
+      end
+
+      def exists?(*keys)
+        super(*keys.map { |key| interpolate(key) })
```
Comment (tubbo) [inline lib/redis/store/namespace.rb:31]: why not just call `exists(*keys)` here?

**Comment by tubbo** [comment]
Can you also rebase from `master` since I just merged #332? Just wanted there to be a clean build that actually tests against the right gem..

File: lib/redis/store/namespace.rb (hunk 1 above)
Comment (byroot) [reply lib/redis/store/namespace.rb:31]: Because they are two different methods, `exists?` being basically `exist() > 0`

Comment (tubbo) [reply lib/redis/store/namespace.rb:31]: oh i see, they have different return types...got it.

**Comment by casperisfine** [comment]
@tubbo rebase done!

File: lib/redis/store/interface.rb [hunk 2]
```diff
@@ -5,8 +5,16 @@ def get(key, options = nil)
         super(key)
       end
 
+      REDIS_SET_OPTIONS = %i(ex px nx xx keepttl).freeze
+      private_constant :REDIS_SET_OPTIONS
```
Comment (tubbo) [inline lib/redis/store/interface.rb:9]: TIL!

**Comment by tubbo** [comment]
Thanks for all your help @casperisfine as well as your maintenance of the Redis gem!

File: lib/redis/store/interface.rb (hunk 2 above)
Comment (PikachuEXE) [reply lib/redis/store/interface.rb:9]: Rubocop got a cop for it
https://docs.rubocop.org/rubocop/cops_style.html#styleconstantvisibility

**Pull request merged**